src/client/common/process/internal/scripts/index.ts

```diff
@@ -49,7 +59,7 @@ export type InterpreterInfoJson = {
 
 export function interpreterInfo(): [string[], (out: string) => InterpreterInfoJson] {
     const script = path.join(SCRIPTS_DIR, 'interpreterInfo.py');
-    const args = [ISOLATED, script];
+    const args = getUseIsolationSetting() ? [ISOLATED, script] : [script];
```

Comment:
It would be cleaner if we avoid duplicating the args:

```suggestion
    const args = [ISOLATED, script];
    if (getUseIsolationSetting()) {
        args.splice(0, 0, ISOLATED);
    }
```

or using a helper:

```suggestion
    const args = maybeIsolated([script]);
```

(The same applies throughout this PR.)